app/models/solution.rb

```diff
@@ -107,10 +107,10 @@ def update_published_iteration_head_tests_status!(status)
   end
 
   def update_latest_iteration_head_tests_status!(status)
-    return if latest_iteration_head_tests_status == status.to_sym
-
-    update_column(:latest_iteration_head_tests_status, status)
-    Solution::SyncToSearchIndex.defer(self)
+    unless latest_iteration_head_tests_status == status.to_sym
+      update_column(:latest_iteration_head_tests_status, status)
+      Solution::SyncToSearchIndex.defer(self)
+    end
     Solution::AutoUpdateToLatestExerciseVersion.(self)
```

Comment:
I think I hit this problem when I tested things locally. I had already put `puts` statements around these lines :)

Reply:
:D!

Nice! System test caught something 🦜